### ISSUE TITLE
1.13/Build 52: Blank Firework Rockets Don't Boost Elytra

### ISSUE THREAD

**Issue by hoorigan** [PaperMC/Paper#1287]
### What behaviour is expected:
USEing a blank firework rocket (no star) while gliding with Elytra should cause a speed boost.

### What behaviour is observed:
The sound effect for the firework rocket plays when USEd, but there is no increase in velocity, nor does there appear to be a particle trail.

### Steps/models to reproduce:
1. Put on elytra.
2. Jump off of something high, go into a glide.
3. Equip blank firework rocket in hand, USE.
4. Be sad and disappointed.  :P

### Plugin list:
None.

### Paper build number:
paperclip-52.jar

### Anything else:
The boost still works if I add a star to my firework rocket.  However, that has the undesired side effect of hurting when the rocket explodes.

**Comment by aikar** [comment]
can you test this in vanilla and spigot? this sounds like it shouldn't of been providing a boost in first place.

**Comment by hoorigan** [comment]
I can confirm that this is currently working in vanilla.  I apologize, I should have said that earlier.

This has worked since 1.11.1 in vanilla, and is currently working in Paper-1261 (MC 1.12.2).

I'll need a few minutes to build a Spigot jar and test it there for 1.13.

**Comment by hoorigan** [comment]
I have confirmed that this is working in Spigot for 1.13.  Here is the version information:

CraftBukkit version git-Spigot-69774b3-3b8f5be (MC: 1.13) (Implementing API version 1.13-R0.1-SNAPSHOT)

This is a freshly built Spigot-1.13.jar.

**Comment by aikar** [comment]
@BillyGalbreath can you take a look at this.

**Comment by aikar** [comment]
really not seeing how the elytra event would be source though, but thats only thing i can think of new to 1.13

**Comment by qixils** [comment]
can't reproduce on paper 49. was something about elytras changed between 49 and 52?

**Comment by hoorigan** [comment]
I have a theory I need to test.

If I understand correctly, Paper fetches and patches a fresh Spigot jar on
startup.  Is that right?

This may have been an upstream issue that was fixed this morning.  I'll
test in a few hours and report back.

**Comment by BillyGalbreath** [comment]
Works fine for me on build 52..

https://youtu.be/2TBVxFk19b4

**Comment by willies952002** [comment]
**^^ Loud Music Warning ^^**

**Comment by AJMFactsheets** [comment]
I've had this problem with 1.12.2 builds of paper. It was because I logged in with entity id 0. It still is a vanilla bug, see [Firework boosting doesn't work in the first world loaded after a game start / if player has entity id 0](https://bugs.mojang.com/browse/MC-111480)

**Comment by aikar** [comment]
Well thats silly, such a simple fix lol. Will fix in 1.12 and 1.13

**Comment by hoorigan** [comment]
Updating to the new build (Paper-64) didn't fix this for me.  I tried removing/replacing the world, as well, and had no change.

I stupidly left out that the server is behind a Waterfall proxy.

I'm not sure why, but updating Waterfall from Waterfall-210 to Waterfall-212 fixed the issue for me.  It could have been simply a matter of restarting Waterfall, or maybe something was fixed between those two versions?

Anyway, seems to be all better now.
